kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/RuntimeDataServiceBase.java

```diff
@@ -289,6 +290,18 @@ public NodeInstanceList getProcessInstanceHistory(long processInstanceId, Boolea
         return nodeInstanceList;
     }
 
+    public NodeInstanceList getProcessInstanceFullHistoryByType(long processInstanceId, String entryType, Integer page, Integer pageSize) {
+
+        logger.debug("About to search for node instances with page {} and page size {}", page, pageSize);
+        Collection<NodeInstanceDesc> result = null;
+
+        result = runtimeDataService.getProcessInstanceFullHistoryByType(processInstanceId, EntryType.valueOf(entryType), buildQueryContext(page, pageSize));
+
+        NodeInstanceList nodeInstanceList = convertToNodeInstanceList(result);
+        logger.debug("Returning result of node instances search: {}", nodeInstanceList);
```

Comment:
Not really useful as it will print out memory address only - no `toString` method is implemented in `NodeInstanceList` class

Reply:
done